File: core/Core/Models/Customer.cs
using System;
using System.Collections.Generic;
using DateOnly = global::System.DateOnly;

namespace Core.Models;

public partial class Customer
{
    public int CustomerId { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public string Email { get; set; } = null!;

    public required string Phone { get; set; }

    public required System.DateOnly BirthDate { get; set; }

    public required DateTime CreatedAt { get; set; }

    public virtual ICollection<Account> Accounts { get; set; } = new List<Account>();

    public virtual ICollection<CustomerAddress> CustomerAddresses { get; set; } = new List<CustomerAddress>();

    public virtual ICollection<Deposit> Deposits { get; set; } = new List<Deposit>();

    public virtual ICollection<Loan> Loans { get; set; } = new List<Loan>();

    public virtual ICollection<LoginLog> LoginLogs { get; set; } = new List<LoginLog>();

    public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();

    public virtual ICollection<PaymentTemplate> PaymentTemplates { get; set; } = new List<PaymentTemplate>();

    public virtual UserCredential? UserCredential { get; set; }
}

File: analytic/domain/customer/customer.py
from datetime import datetime, date


class Customer:
    def __init__(self,
                 customer_id: int,
                 first_name: str,
                 last_name: str,
                 email: str,
                 phone: str | None,
                 birth_date: date | None,
                 created_at: datetime | None):
        self.customer_id = customer_id
        self.first_name = first_name
        self.last_name = last_name
        self.email = email
        self.phone = phone
        self.birth_date = birth_date
        self.created_at = created_at
